pkg/controller/attach_detach.go

```diff
@@ -282,6 +287,17 @@ func (r *ReconcileAttachDetach) triggerAttach(ctx context.Context, azVolumeAttac
 						r.eventRecorder.Eventf(pod.DeepCopyObject(), v1.EventTypeNormal, consts.ReplicaAttachmentSuccessEvent, "Replica mount for volume %s successfully attached to node %s", azVolumeAttachment.Spec.VolumeName, azVolumeAttachment.Spec.NodeName)
 					}
 				}
+				// the node's remaining capacity of disk attachment should be decreased by 1, since the disk attachment is succeeded.
+				remainingCapacity, nodeExists := r.SharedState.nodeDiskAvailabilityMap.Load(azVolumeAttachment.Spec.NodeName)
+				if nodeExists && remainingCapacity != nil {
+					currentCapacity := int32(0)
+					for {
+						currentCapacity = remainingCapacity.(*atomic.Int32).Load()
+						if currentCapacity == int32(0) || remainingCapacity.(*atomic.Int32).CompareAndSwap(currentCapacity, currentCapacity-1) {
+							break
```

Comment:
We should probably log an error(?) if the remaining capacity was already zero.

Reply:
Done.